src/test/unit/io/reader_test.cpp

```diff
@@ -1,6 +1,29 @@
 #include <gtest/gtest.h>
 #include <stan/io/reader.hpp>
-
+#include <tuple>
+
+namespace stan {
+namespace test {
+auto get_sparse_matrix_reader_pars() {
+  int row_size = (rand() % 20) + 1;
```

Comment:
I would prefer to see the tests iterate over some well; chose row and column sizes that exercise the boundaries, like

```
std::vector<int> idxs{0, 1, 2, 5, 10};
for (row_size : idxs)
  for (col_size : idxs) 
    ... test ...
```